etcd/etcd-live-cluster-reconfiguration.md

```diff
@@ -1,236 +1,350 @@
 
-## Change etcd cluster size
+## Configuring etcd using Container Linux Config
 
-When [a Container Linux Config][cl-configs] is used to configure an etcd member on a Container Linux node, it compiles a special `/etc/systemd/system/etcd-member.service.d/20-clct-etcd-member.conf` [drop-in unit file][drop-in]. That is, the Container Linux Config below:
+When a [Container Linux Config][cl-configs] is used for configuring an etcd member on a Container Linux node, it compiles a special `/etc/systemd/system/etcd-member.service.d/20-clct-etcd-member.conf` [drop-in unit file][drop-in]. For example:
 
 ```yaml container-linux-config
 etcd:
+  name: demo-etcd-1
+  listen_client_urls: https://10.240.0.1:2379,http://0.0.0.0:4001
   advertise_client_urls: http://<PEER_ADDRESS>:2379
+  listen_peer_urls:            http://0.0.0.0:2380
   initial_advertise_peer_urls: http://<PEER_ADDRESS>:2380
-  listen_client_urls: http://0.0.0.0:2379,http://0.0.0.0:4001
-  listen_peer_urls: http://0.0.0.0:2380
-  discovery: https://discovery.etcd.io/<token>
+  initial_cluster:             demo-etcd-1=https://0.0.0.1:2380,demo-etcd-2=https://0.0.0.2:2380,demo-etcd-3=https://0.0.0.3:2380
+  initial_cluster_token:       demo-etcd-token
+  initial_cluster_state:       new
 ```
 
-will generate the following [drop-in][drop-in]:
+The config file is first validated and transformed into a machine-readable form, which is then sent directly to a Container Linux provisioning target. The [drop-in][drop-in] generated from the example config file is given below:
 
 ```ini
 [Service]
-Environment="ETCD_IMAGE_TAG=v3.1.4"
 ExecStart=
 ExecStart=/usr/lib/coreos/etcd-wrapper $ETCD_OPTS \
-  --advertise-client-urls: http://<PEER_ADDRESS>:2379 \
-  --initial-advertise-peer-urls: http://<PEER_ADDRESS>:2380 \
-  --listen-client-urls: http://0.0.0.0:2379,http://0.0.0.0:4001 \
-  --listen-peer-urls: http://0.0.0.0:2380 \
-  --discovery: https://discovery.etcd.io/<token>
+  --name="demo-etcd-1" \
+  --listen-peer-urls="http://0.0.0.0:2380" \
+  --listen-client-urls="https://10.240.0.1:2379,http://0.0.0.0:4001" \
+  --initial-advertise-peer-urls="http://<PEER_ADDRESS>:2380" \
+  --initial-cluster="demo-etcd-1=https://0.0.0.1:2380,demo-etcd-2=https://0.0.0.2:2380,demo-etcd-3=https://0.0.0.3:2380" \
+  --initial-cluster-state="new" \
+  --initial-cluster-token="demo-etcd-token" \
+  --advertise-client-urls="http://<PEER_ADDRESS>:2379"
 ```
 
-If the etcd cluster is secured with TLS, use `https://` instead of `http://` in the command examples below.
+If the etcd cluster is secured with TLS, use `https://` instead of `http://` in the config files. If the peer addresses for the initial cluster are known at time of spinning up the cluster, `--discovery="https://discovery.etcd.io/<token>` is not required.
```

Comment:
`--discovery` isn't given so this is a nonsequitor. Instead: If the peer addresses for the initial cluster are unknown when provisioning the cluster, use the etcd discovery service with `--discovery="https://discovery.etcd.io/<token>`.